### ISSUE TITLE
Make "last activity" notices on question lists more useful

### ISSUE THREAD

**Issue by cellio** [codidact/qpixel#315]
https://meta.codidact.com/questions/279509
https://meta.codidact.com/q/277221

The question list shows a generic "last activity" notice, with time and who did it.  But it's hard to see where that happened or what the activity was.  On a question with several answers you have to page through looking for an edit or a new answer, and it might actually be that something was deleted or the question was closed (or opened).

Two requests -- either alone would be an improvement, and both together would add a lot of value:

1. Be more specific in the entry on the question list -- "edited", "answered", "closed", "answer deleted" would all be more specific than "last activity".  (There's no need for a notice for the *question* being deleted; if it has been, it won't be on the question list.)

2. Make the notice a link to the post that was affected.

I suspect this is hard because we would have done it already if it were easy; these aren't the first requests to improve this aspect of the interface.  But I don't know what's involved, so I'm tagging it "unassessed" so someone else can evaluate.

**Comment by cellio** [comment]
The post list now shows "edited", "closed", "closed as duplicate", and "asked".  New answers and edits to answers both still produce "last activity" -- would it be possible for those to be "answered" and "answer edited"?  Those are the common remaining use cases; it'd be great if we could catch them.

I believe that leaves only "answer deleted".  I think it's ok to leave that as "last activity"; most people can't see deleted answers anyway, so announcing a deletion might just raise curiosity that can't be satisfied.  On the other hand, it'd be a little more transparent.  We could go either way on that one.

This issue also asks for these notices to be links to the affected post; we can spin that off into a separate issue if it that would be easier to manage.

**Comment by cellio** [comment]
Related: https://github.com/codidact/qpixel/issues/385

**Comment by cellio** [comment]
There are some suggestions in https://github.com/codidact/qpixel/issues/385 (which I'm closing as a duplicate of this issue).

**Comment by trichoplax** [comment]
I raised a [related Meta post about the grey circle new activity indicators on the Category tabs](https://meta.codidact.com/posts/287712) that can be confusing when a *question* is deleted.

I agree that deleted questions don't need consideration here because they won't show up on the question list page, but they may need consideration in deciding when to show the grey circle. Otherwise the user clicks to see what the grey circle is about and finds a question list with no recent activity. Should that be part of this issue or a separate one?

**Comment by trichoplax** [comment]
As for deleted *answers*, it may cause more frustrating curiosity to have "last activity 2 hours ago" with no visible activity on the question or any of its answers, than to have the more specific "answer deleted 2 hours ago" where at least the user then knows that their curiosity cannot be satisfied so they don't waste effort going through all the answers.

**Comment by cellio** [comment]
Agreed -- we want "answer deleted" rather than "last activity", so people who can't see deleted answers at least know what the issue was.

The problem with deleted *questions* causing the category to update (so you get the silver dot) sounds like it's in a different part of the code, so let's create a separate issue for that.

**Comment by trichoplax** [comment]
Separate issue created: #952

**Comment by Taeir** [comment]
In my opinion a deletion of a question or answer should not trigger last_activity at all. I think it's very annoying how posts get moved to the top because of "activity" when that activity was the removal of something (though perhaps its useful for moderators?).
At the same time, it feels wrong to have a "last_activity" timestamp and not update it for deletions.

**Comment by cellio** [comment]
Would it complicate the code too much to check if the user can see deleted posts?  _If_ you can see them then there's something to audit, so it's reasonable to show it (lets people notice if someone's going around deleting good content).  If you can't see deleted posts, though, then there's nothing to see or do, so don't update.  Would that work?  Or do we cache post lists in a user-independent way and making it user-dependent would be disruptive?

**Comment by trichoplax** [comment]
If there is a problem with user-dependent caching (no idea if there is - I haven't looked at the code):

Since the deletion of a post isn't secret, just unhelpful to display to a user who can't see deleted posts, could we keep track of both the most recent change (even if it's a deletion) and the most recent non-deletion change (which in some cases will be the same change)?

Then we can access whichever one of these two matches the user in question.
